src/prefect/utilities/visualization.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any
+import graphviz
+
+
+class FlowVisualizationError(Exception):
+    pass
+
+
+class TaskRunTrackerState:
+    current = None
+
+
+def get_task_run_tracker():
+    return TaskRunTrackerState.current
+
+
+def track_task_run(task_name, parameters, mock_return=None):
+    task_run_tracker = get_task_run_tracker()
+    if task_run_tracker:
+        upstream_tasks = []
+        for k, v in parameters.items():
+            if isinstance(v, TrackableTask):
+                upstream_tasks.append(v)
+
+        trackable_task = TrackableTask(
+            name=task_name,
+            upstream_tasks=upstream_tasks,
+            value=mock_return,
+        )
+        task_run_tracker.add_trackable_task(trackable_task)
+        return trackable_task
+
+
+class TrackableTask:
+    def __init__(
+        self,
+        name: str,
+        upstream_tasks: list["TrackableTask"],
+        value: Any = None,
+    ):
+        self.name = name
+        self.upstream_tasks = upstream_tasks
+        self.value = value
+
+    def __iter__(self):
```

Comment:
this solves the very specific iter case, but is not good enough when there is all the other `__xxx__` methods to support. looking into https://wrapt.readthedocs.io/en/latest/ and proxy objects